helm-chart/renku/templates/_helpers.tpl

```diff
@@ -103,3 +103,47 @@ fail "External PostgreSQL and Renku-bundled PostgreSQL cannot both be disabled.
 {{- if .Values.global.externalServices.postgresql.enabled and .Values.global.externalServices.postgresql.password and .Values.global.externalServices.postgresql.existingSecret -}}
 fail "External PostgreSQL password and existing Secret fields cannot both be populated."
 {{- end -}}
+
+{{- define "keycloak.admin-secret" -}}
+{{- $secretAdmin := lookup "v1" "Secret" .Release.Namespace "keycloak-password-secret" -}}
+{{- if $secretAdmin -}}
+{{- if $secretAdmin.data.KEYCLOAK_ADMIN -}}
+# Post-keycloakx
+KEYCLOAK_ADMIN: {{ $secretAdmin.data.KEYCLOAK_ADMIN | quote }}
+KEYCLOAK_ADMIN_PASSWORD: {{ $secretAdmin.data.KEYCLOAK_ADMIN_PASSWORD | quote }}
+{{- else -}}
+# Pre-keycloakx
+KEYCLOAK_ADMIN: {{ $secretAdmin.data.KEYCLOAK_USER | quote }}
+KEYCLOAK_ADMIN_PASSWORD: {{ $secretAdmin.data.KEYCLOAK_PASSWORD | quote }}
+{{- end -}}
+# No pre-existing secret
+{{- else -}}
+KEYCLOAK_ADMIN: {{ .Values.global.keycloak.user | b64enc }}
+KEYCLOAK_ADMIN_PASSWORD: {{ default (randAlphaNum 64) .Values.global.keycloak.password.value | b64enc }}
+{{- end -}}
+{{- end -}}
+
+{{- define "keycloak.postgres-secret" -}}
+{{- $secretPostgres := lookup "v1" "Secret" .Release.Namespace "renku-keycloak-postgres" -}}
+{{- if $secretPostgres -}}
+{{- if $secretPostgres.data.KC_DB_URL_HOST -}}
```

Comment:
Same as the other helper section. I suspect this may crash the helm chart if the key is not present and the else statement will never be reached.